src/context/AuthContext.tsx

```diff
@@ -1,28 +1,95 @@
 import { createContext, useState, useEffect, ReactNode } from "react";
 import {
   type User,
-  type AuthContext as AuthContextType,
+  type AuthContextType,
+  type LoginCredentials,
+  type SignupCredentials,
 } from "../interfaces/types";
-import { getCurrentUser as getCurrentUserApi } from "../utils/auth";
+import { authApi } from "../services/authServices";
 
-// Create the context
 export const AuthContext = createContext<AuthContextType | undefined>(
   undefined
 );
 
-// Create the Provider Component
-export const AuthProvider = ({ children }: { children: ReactNode }) => {
-  const [user, setUser] = useState<User | null>(null);
+export const AuthProvider: React.FC<{ children: ReactNode }> = ({
+  children,
+}) => {
+  const [user, setUser] = useState<User | null>(() => {
+    const storedUser = localStorage.getItem("user");
+    return storedUser ? JSON.parse(storedUser) : null;
+  });
+  const [isLoading, setIsLoading] = useState(true);
+
+  // Get the information of the current user
+  const fetchUser = async () => {
+    setIsLoading(true);
+    try {
+      const response = await authApi.getCurrentUser(); // send api request to get the current user
+      if (response) {
+        setUser(response);
+        localStorage.setItem("user", JSON.stringify(response)); // store the user info in local storage
+      } else {
+        setUser(null);
+        localStorage.removeItem("user");
+      }
+    } catch (error) {
+      console.log("Error fetching user: ", error);
```

Comment:
create a util that logs only in development

Reply:
I added the logger dai, it works only in the dev environment.